lighthouse-core/gather/gather-runner.js

```diff
@@ -626,7 +626,18 @@ class GatherRunner {
    */
   static finalizeBaseArtifacts(baseArtifacts) {
     // Take only unique LighthouseRunWarnings.
-    baseArtifacts.LighthouseRunWarnings = Array.from(new Set(baseArtifacts.LighthouseRunWarnings));
+
+    const uniqueWarnings = new Set();
+    const LighthouseRunWarnings = [];
+    for (const warning of baseArtifacts.LighthouseRunWarnings) {
+      // each entry in baseArtifacts.LighthouseRunWarnings can be a string or a LH.IcuMessage.
+      const stringifiedWarning = JSON.stringify(warning);
```

Comment:
We have the `lodash.isequal` package in our dependencies which does deep equality like what we're looking for here and isn't sensitive to ordering like a stringify approach is. 

Given we tend to have very few warnings here, I'm not as concerned with preserving hashset time complexity :)

WDYT about switching to a `LighthouseRunWarnings.some(existing => isDeepEqual(warning, existing)` check?

Reply:
i see what you mean. updating the branch